### PR TITLE
Complete Maps API for MVP

### PR DESCRIPTION
**Pull request by yjzhang111**
Finish Maps API for MVP, including Geolocation API, which asks for user location, and Places API, which gets nearby coffee shop for users. Delete hard-coded markers for the prototype.

### CODE REVIEW

**Review by TylerHernandez — APPROVED** [review]

**Review by dag13 — APPROVED** [review]

File: src/main/webapp/map.js [hunk 1]
```diff
@@ -1,43 +1,85 @@
...
+/** Creates a map that shows all coffee shops around the user. */
 function createMap() {
+  // Set default location at new york city
+  var newyork = new google.maps.LatLng(40.7128, -74.0060);
   map = new google.maps.Map(
     document.getElementById('map'),
-    {center: {lat: 40.7128, lng: -74.0060}, zoom: 12});
+    {center: newyork, zoom: 14});
+  
+  // Infowindow for to handle error in get user location
```
Comment (mutisog) [inline src/main/webapp/map.js:14]: typo

File: src/main/webapp/map.js [hunk 2]
```diff
@@ -1,43 +1,85 @@
...
+        userPos = {
+          lat: position.coords.latitude,
+          lng: position.coords.longitude
+        };
+        map.setCenter(userPos);
+
+        // Request nearby coffee shop info
+        var request = {
+          location: userPos,
+          radius: '500',
```
Comment (mutisog) [inline src/main/webapp/map.js:32]: where are the units for this radius specified?

File: src/main/webapp/map.js [hunk 3]
```diff
@@ -1,43 +1,85 @@
...
+    navigator.geolocation.getCurrentPosition(
+      function(position) {
+        userPos = {
+          lat: position.coords.latitude,
+          lng: position.coords.longitude
+        };
+        map.setCenter(userPos);
+
+        // Request nearby coffee shop info
+        var request = {
```
Comment (mutisog) [inline src/main/webapp/map.js:30]: consider more descriptive name e.g. getCoffeeShopsRequest

File: src/main/webapp/map.js [hunk 4]
```diff
@@ -1,43 +1,85 @@
...
+  infoWindow.setPosition(userPos);
+  infoWindow.setContent(
+    browserHasGeolocation
+      ? "Error: The Geolocation service failed."
+      : "Error: Your browser doesn't support geolocation."
+  );
+  infoWindow.open(map);
+}
+
+function callback(results, status) {
```
Comment (mutisog) [inline src/main/webapp/map.js:58]: what does the 'results' parameter contain?

File: src/main/webapp/map.js [hunk 5]
```diff
@@ -1,43 +1,85 @@
...
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(
+      function(position) {
+        userPos = {
+          lat: position.coords.latitude,
+          lng: position.coords.longitude
+        };
+        map.setCenter(userPos);
+
+        // Request nearby coffee shop info
```
Comment (mutisog) [inline src/main/webapp/map.js:29]: Possible stretch goal: allow a user to specify their desired search radius.

File: src/main/webapp/map.js [hunk 6]
```diff
@@ -1,43 +1,85 @@
...
-/** 
- *  Creates a marker that shows a info window when clicked. 
- *  The link leads to the coffee shop page which contains
- *  individual coffee shop information
- */
-function createMarkerForDisplay(shop) {
-  const marker =
-    new google.maps.Marker({position: {lat: shop[0], lng: shop[1]}, map: map, title: shop[2]});
-  const infoWindow = new google.maps.InfoWindow({content: '<a href=coffeeshop.html>' + shop[2] + '</a>'});
+function createMarker(place) {
```
Comment (mutisog) [inline src/main/webapp/map.js:66]: document what this function does. In general, if a function's purpose isn't super easy to understand, and the name doesn't make it obvious, adding a short comment aids readability.

File: src/main/webapp/map.js [hunk 7]
```diff
@@ -1,43 +1,85 @@
...
+var userPos;
 
-/* Editable marker that displays when a user clicks in the map. */
-let editMarker;
-
-let markers = [[40.7122879, -74.0106218, "Starbucks"], [40.7105991, -74.0112465, "Irving Farm New York"]]
-
-/** Creates a map that allows users to add markers. */
+/** Creates a map that shows all coffee shops around the user. */
 function createMap() {
```
Comment (mutisog) [inline src/main/webapp/map.js:7]: Do we have tests for this functionality?

**Review by mutisog — COMMENTED** [review]
Added some comments.

File: src/main/webapp/map.js (hunk 2 above)
Comment (yjzhang111) [reply src/main/webapp/map.js:32]: Added a comment specifying the unit (metres)

File: src/main/webapp/map.js (hunk 4 above)
Comment (yjzhang111) [reply src/main/webapp/map.js:58]: It contains everything returned by the Places API query in JSON format. (Example here: https://developers.google.com/places/web-service/search#nearby-search-and-text-search-responses)

**Pull request merged**